R/bootstrap.R

```diff
@@ -938,10 +944,10 @@ buildTabset <- function(tabs, ulClass, textFilter = NULL, id = NULL,
             tabs = tabs, textFilter = textFilter)
 
   tabNavList <- tags$ul(class = ulClass, id = id,
-                  `data-tabsetid` = tabsetId, lapply(tabs, "[[", 1))
+                  `data-tabsetid` = tabsetId, !!!lapply(tabs, "[[", "liTag"))
 
   tabContent <- tags$div(class = "tab-content",
-                  `data-tabsetid` = tabsetId, lapply(tabs, "[[", 2))
+                  `data-tabsetid` = tabsetId, !!!lapply(tabs, "[[", "divTag"))
```

Comment:
Should we consider extracting non-tabPanel() content and putting it above/below the `tab-content`?

Reply:
Here's an example that we should think about -- what _should_ happen with code like this?

```R
shinyApp(
  navbarPage(
    "Title",
    div("other stuff 1"),
    tabPanel("a", "tab a"),
    div("other stuff 2"),
    tabPanel("b", "tab b"),
    div("other stuff 3")
  ),
  function(input, output) { }
)
```

Reply:
I decided to revert back to supporting this off-label usage with a warning in https://github.com/rstudio/shiny/pull/3315/commits/2e33634726004fe6f16830c8a34a0effaef5b0eb

(Thankfully this has only ever worked with tags, so we can throw we a better message in any other case)